Makefile.in

```diff
@@ -92,8 +94,46 @@ hacking: ocaml-stage1-config.status stage0 \
 	cp ocaml-stage1-config.status ocaml/config.status
 	(cd ocaml && ./config.status)
 	PATH=$(stage0_prefix)/bin:$$PATH \
+	  ARCH=`grep '^ARCH=' ocaml/Makefile.config | cut -d'=' -f2` \
 	  $(dune) build -w --profile=release --build-dir=_build1 @install
 
+
+## Test compilation of backend-specific parts
+ARCHES=amd64 i386 arm arm64 power s390x riscv
+ARCH_SPECIFIC =\
+  backend/arch.ml backend/proc.ml backend/CSE.ml backend/selection.ml \
+  backend/scheduling.ml backend/reload.ml backend/emit.ml
+
+# This rule provides a quick way to check that machine-dependent
+# files compiles fine for a foreign architecture (passed as ARCH=xxx).
+# Based on stage1 target above.
+.PHONY: check_arch
+check_arch: ocaml-stage1-config.status stage0 \
+	config_files
+	cp ocaml-stage1-config.status ocaml/config.status
+	(cd ocaml && ./config.status)
+	@echo "========= CHECKING backend/$(ARCH) =============="
+	@rm -f $(ARCH_SPECIFIC)
+	@rm -f backend/*.cm* backend/cfg/*.cm* backend/debug/*.cm*
```

Comment:
I don't understand this line -- this seems to be operating on the source directory?

Reply:
You are right. I've removed it. Dune knows to rebuild the backend directory when .ml files change.